plasmapy/particles/particle_class.py

```diff
@@ -2036,7 +2036,111 @@ def charge(self, q: Optional[Union[u.Quantity, Real]]):
+
+Notes
+-----
+Particles in PlasmaPy are canonically represented as instances of the
+`~plasmapy.particles.particle_class.Particle` class.
+
+>>> from plasmapy.particles import Particle
+>>> Particle("proton")
+Particle("p+")
+>>> Particle("neutron")
+Particle("n")
+
+All `~plasmapy.particles.particle_class.Particle` instances are
+particle-like.  Any `object` that can be passed as a sole argument to
+the `~plasmapy.particles.particle_class.Particle` class is considered
+particle-like.
+
+An **element** may also be represented by a string that contains the
+atomic symbol (case-sensitive) or the name of the element, or an integer
+representing the atomic number.  For example, iron can be represented as
+``"Fe"``, ``"iron"``, ``"Iron"``, or ``26``.  If any of these objects is
+passed to `~plasmapy.particles.particle_class.Particle`, it will create
+a `~plasmapy.particles.particle_class.Particle` instance for iron.
+
+An **isotope** may also be represented by a string that contains an
+atomic symbol or element name followed by a hyphen and the mass number
+(with no spaces in between).   For example, :sup:`56`Fe can be
+represented as ``"Fe-56"``, ``"iron-56"``, or ``Particle("Fe-56")``.
+A `~plasmapy.particles.particle_class.Particle` instance for an isotope
+can also be created by passing a representation of iron as the first
+argument and specifying the mass number with the argument ``mass_numb``.
+
+>>> Particle(26, mass_numb=56)
+Particle("Fe-56")
+
+An **ion** or **ionic level** is represented as a string that contains a
+representation of an element or isotope, followed by a space, and then
+the charge information (e.g., ``"2+"``).  The charge information in the
+string includes an integer either followed or preceded by a plus or
+minus sign.  For example, Fe:sup:`2+` can be represented as
+``"Fe 2+"``, while :sup:`56`Fe:sup:`0+` can be represented as
+``"iron-56 +0"``.  The integer charge can alternatively be specified
+using the ``Z`` keyword to `~plasmapy.particles.particle_class.Particle`.
+
+>>> Particle("He-4", Z=2)
+Particle("He-4 2+")
+
+A **special particle** may be represented by a string that contains a
+the name of the particle or a standard symbol for it.  A neutron can be
+represented as ``"n"`` or ``"neutron"``, a proton can be represented as
+``"p"``, ``"p+"``, or ``"proton"``, and an electron can be represented
+by ``"e"``, ``"e-"``, or ``"electron"``.
+
+Instances of the `~plasmapy.particles.particle_class.CustomParticle`
+class are considered particle-like because the mass and charge of the
+custom particle are uniquely defined in physical units.
+
+`~plasmapy.particles.particle_class.DimensionlessParticle` instances are
+not considered particle-like because (without normalization information)
+they do not *uniquely* identify a particle.
+"""
+
+
+def is_particle_like(obj: Any) -> bool:
+    """
+    Return `True` if an `object` can uniquely represent a physical
+    particle, and `False` otherwise.
```

Comment:
I'm still wondering if we want to have `is_particle_like` in here.  I'm tentatively including it since there are times when code like this could end up making things more readable, but I'm open to deleting it.

It would probably be cleaner to make `particle_like` callable (perhaps by setting `particle_like.__call__` to `is_particle_like`), but I ran into some weirdness when trying that.